tests/importer/importer.cpp

```diff
@@ -1106,3 +1106,79 @@ TEST(Importer, basePathNotEndingInSlash)
     importer->resolveImports(importedUnits, resourcePath("importer"));
     EXPECT_FALSE(importedUnits->hasUnresolvedImports());
 }
+
+TEST(Importer, importInvalidXml)
+{
+    auto e = "The attempt to import the model at '" + resourcePath("importer/not_even_proper.xml") + "' failed: the file is not valid XML.";
+
+    auto importer = libcellml::Importer::create();
+    auto parser = libcellml::Parser::create();
+
+    auto model = parser->parseModel(fileContents("importer/import_invalid_xml.cellml"));
+
+    importer->resolveImports(model, resourcePath("importer"));
+    EXPECT_EQ(size_t(1), importer->errorCount());
```

Comment:
random question for @hsorby - in the original commit of the failing tests to demonstrate this problem, you were expecting 3 errors. Was that just a random number or did you expect there to be other errors that became non-errors as you implemented the fix?

Reply:
It was semi-random, I was expecting around three errors but in the implementation once I hit an XML error I stopped and just returned the first one.